strings/longest_substr_no_repeating.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+* @file
+* @brief Sliding window solution to find the longest substring with no repeating characters
+* @details
+    * Given a string we are required to find the
+    * longest possible substring without any repeating characters.
+    * The time and space complexity are both O(n) since an unordered set is used
+    * and the entire string is traversed.
```

Comment:
```suggestion
 * Given a string we are required to find the
 * longest possible substring without any repeating characters.
 * The time and space complexity are both O(n) since an unordered set is used
 * and the entire string is traversed.
```